neutron/services/loadbalancer/drivers/common/agent_driver_base.py

```diff
@@ -85,8 +86,12 @@ def get_logical_device(self, context, pool_id=None):
         with context.session.begin(subtransactions=True):
             qry = context.session.query(loadbalancer_db.Pool)
             qry = qry.filter_by(id=pool_id)
-            pool = qry.one()
             retval = {}
+            # qry.one will raise NoResultFound if pool not exist
+            try:
+                pool = qry.one()
+            except exc.NoResultFound:
+                return retval
```

Comment:
这个改动是跟#35 有关系， 但是代码在server端，所有我就把它放在这个commit了， 你把#35 revert掉把。
我把这个改动加到#35这个request中

2016-06-27 19:26 GMT+08:00 huntxu <notifications@github.com>:

> In neutron/services/loadbalancer/drivers/common/agent_driver_base.py
> <https://github.com/eayunstack/neutron/pull/36#discussion_r68560697>:
>
> >              retval = {}
> > +            # qry.one will raise NoResultFound if pool not exist
> > +            try:
> > +                pool = qry.one()
> > +            except exc.NoResultFound:
> > +                return retval
>
> 这处改动这整个commit关系不大
> 如果是为之前的 #35 <https://github.com/eayunstack/neutron/pull/35>
> 而改动的话，通知我一声revert掉 #35 <https://github.com/eayunstack/neutron/pull/35>
> 把这个改动加进去
> 如果是解决另外的bug则另外在 redmine 上打开issue然后提交独立的pull request解决，不要混在这个commit之中
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eayunstack/neutron/pull/36/files/6ff5a7a848144ca38a3fda1b022d0f2b431c880b#r68560697>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGUT0cHJg9mtVlvEWnq-sp-iV8nH7S1kks5qP7NogaJpZM4I8a7c>
> .
>